File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/QueueStreamEvent.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.room.RoomDataModel;

public interface QueueStreamEvent {
    void onEnterQueue(QueueStreamEventController c);
    void onBattleFound(RoomDataModel r);
    void onBattleNotFound();
    void onJoinWaitingRoom();
    void onDisconnected();
    void onError(String s);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/QueueStreamEvent.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.room.RoomDataModel

interface QueueStreamEvent {
    fun onEnterQueue(c : QueueStreamEventController)
    fun onBattleFound(r : RoomDataModel)
    fun onBattleNotFound()
    fun onJoinWaitingRoom()
    fun onLeftWaitingRoom()
    fun toBattle()
    fun onDisconnected()
    fun onError(s : String)
}